include/CLI/App.hpp

```diff
@@ -523,133 +536,157 @@ class App {
     }
 
     /// Set a help flag, replace the existing one if present
-    Option *set_help_flag(std::string flag_name = "", std::string help_description = "") {
+    Option *set_help_flag(std::string flag_name = "", const std::string &help_description = "") {
+        // take flag_description by const reference otherwise add_flag tries to assign to help_description
         if(help_ptr_ != nullptr) {
             remove_option(help_ptr_);
             help_ptr_ = nullptr;
         }
 
         // Empty name will simply remove the help flag
         if(!flag_name.empty()) {
-            help_ptr_ = add_flag(flag_name, std::move(help_description));
+            help_ptr_ = add_flag(flag_name, help_description);
```

Comment:
it uses a const std::string & now, so it makes sure to trigger the correct overload with the description and description has to be const otherwise it will try to write to it.    So you can't move a const std::string &

Reply:
Okay, makes sense.